connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -43,10 +43,38 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeoutException;
 
-public class RestClient {
+public class RestClient implements AutoCloseable {
```

Comment:
Should we add a note on thread-safety to this class, since we're creating and sharing a single instance per worker?

Also, have you checked and verified that this class is actually thread-safe?

Reply:
Yes I looked through the mutability paths for this class and believe it to be thread safe, and added a comment to that effect.

Reply:
Do we know that the components we use (such as the SSL factory and the Jetty `HttpClient`) are themselves thread-safe?

Reply:
It looks like we've elected to create a new Jetty `HttpClient` with every REST request; were we not able to verify the thread-safety of one or more components?

Reply:
I wasn't able to verify the thread-safety of the HttpClient in all versions of Jetty, so I think i'll be leaving the existing pattern of one-restclient-per-request in place.

Reply:
Sounds good. This shouldn't be any less performant than the current `trunk` and we can always revisit later.